### ISSUE TITLE
Unable to run XGBoost in R - Invalid Parameter format for nthread expect int but value='4'

### ISSUE THREAD

**Issue by ianlokh** [dmlc/xgboost#4423]
Hi,

I keep encountering the following error after building XGBoost for R

Error in xgb.iter.update(bst$handle, dtrain, iteration - 1, obj) : 
  Invalid Parameter format for nthread expect int but value='4'

Here is my code:
```
library(xgboost)
x = matrix(rnorm(1000*10000), 10000, 100)
y = x %*% rnorm(100) + rnorm(1000)

system.time({bst = xgboost(data = x, label = y, nthread = 4, nround = 100, verbose = FALSE)})
```

I have tried re-compiling the source but I am still getting the same error. I do not encounter this error when I install the package from CRAN

**Comment by hcho3** [comment]
@ianlokh I cannot reproduce the problem. Can you post details about your system (OS, locale, C compiler version etc)?

**Comment by ianlokh** [comment]
Hi Philip, thanks for the reply and followup! Previously I was on R 3.5.2 (OSX) and I upgraded to R 3.6.0 and recompiled XGBoost for R and somehow I did not get the error anymore. I suspect it could be some conflicts with other packages that I installed in 3.5.2.

This is because after installing 3.6.0 in OSX I had a clean library and the first package I tried to install was XGBoost and was able to successfully run it in multi-threaded mode.

Since this is no longer an issue for me, I will close this ticket.

Thank you for your kind help and time in investigating this problem.

Best Regards
Ian
